llvm/lib/Transforms/Vectorize/VPlanAnalysis.h

```diff
@@ -67,6 +69,15 @@ class VPTypeAnalysis {
 // Collect a VPlan's ephemeral recipes (those used only by an assume).
 void collectEphemeralRecipesForVPlan(VPlan &Plan,
                                      DenseSet<VPRecipeBase *> &EphRecipes);
+
+namespace vputils {
+
+/// Return the SCEV expression for \p V. Returns SCEVCouldNotCompute if no
+/// SCEV expression could be constructed.
+const SCEV *getSCEVExprForVPValue(VPValue *V, ScalarEvolution &SE);
```

Comment:
Does this more naturally belong to VPlan.h and VPlan.cpp, alongside other vputils functions, in particular the complementary vputils::getOrCreateVPValueForSCEVExpr()?

Reply:
`VPlan.h` has grown quite a bit by now, so I think it would possibly make sense to split things up by functionality a bit (same for VPlan.cpp). it doesn't necessarily have to go into `VPlanAnalysis.cpp` but could be moved into a separate VPlanUtils.h as an alternative.

Reply:
Ok, +1 to reorganize all `vputils` functions in files of their own.

Most of these would presumably promote to a type system of VPValues, or other refinements like introducing an extract-first-lane as only user to capture `onlyFirstLaneUsed()`.

Reply:
Moved to VPlanUtils.cpp